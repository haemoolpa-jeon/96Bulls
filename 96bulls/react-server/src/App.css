.home-page {
  text-align: center;
  background-color: #9919DD;
  width:100%;
  height: 100%;
  margin-top: 5%;
  display: flex;
  align-items: center;
}

#edit-profile-page {
  height: 100%;
  padding: 100px;
  display: flex;
  justify-content: center;
}

.shape-container {
  border: 2px solid black;
  border-radius: 20px;
  height: 300px;
  margin-top: 30px;
  padding: 50px;
  width: 60%;
}

.single-shape-container {
  display: flex;
  flex-direction: column;
  justify-content: center;
  padding: 0 25px;
  text-align: center;
  cursor: pointer;
}

.selected {
  background-color: blue !important;
}

.shape-caption {
  font-style: italic;
  margin-top: 5px;
}

#circle {
  height: 50px;
  width: 50px;
  background-color: green;
  border-radius: 50%;
  display: block;
}

#triangle {
  width: 0;
  height: 0;
  border-left: 25px solid transparent;
  border-right: 25px solid transparent;
  border-bottom: 50px solid green;
  display: block;
}

#square {
  width: 50px;
  height: 50px;
  background: green;
}

#rectangle {
  height: 50px;
  width: 100px;
  background-color: green;
  display: block;
}

.muted {
  color: #8d8d8d;
}

.flex-center {
  display: flex;
  align-items: center;
  justify-content: center;
}

.flex-bottom-right {
  display: flex;
  flex-direction: column;
  align-items: flex-end;
  justify-content: flex-end;
}

.App-logo {
  height: 40vmin;
  pointer-events: none;
}

@media (prefers-reduced-motion: no-preference) {
  .App-logo {
    animation: App-logo-spin infinite 20s linear;
  }
}

.home-card {
  /* background-color: #282c34; */
  min-height: 50vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: space-between;
  font-size: calc(10px + 2vmin);
  background-color: white;
  width: 60vh;
  margin: auto;
}

.current-class-page {
  background-color: rgb(112, 112, 112);
  height: 100%;
}

.back-button {
  padding: 20px;
}

.back-button a {
  margin-top: 20px;
  text-decoration: none;
  color: black;
}

.back-button:hover {
  cursor: pointer;
}

.info-cards {
  background-color: #bbbbbb;
  color: white;
  text-align: center;
  padding: 15px 20px;
  margin-bottom: 10px;
  font-size: 20px;
  font-style: italic;
}

.page-header {
  padding-top: 40px;
  font-size: 30px;
  font-weight: 500;
  color: white;
}

.page-subheader {
  color: white;
  font-size: 20px;
  margin-top: 20px;
}

.user-levels-section {
  padding-top: 50px;
  display: grid;
  grid-template-columns: repeat(7, 1fr);
  grid-row-gap: 20px;
}

.level-obj-container {
  display: flex;
  justify-content: center;
  align-items: center;
  flex-direction: column;
  color: white;
  font-style: italic;
  font-size: 15px;
}

.level-circle {
  background-color: white;
  color: black;
  height: 58px;
  width: 58px;
  border-radius: 100%;
  display: flex;
  justify-content: center;
  align-items: center;
  font-size: 15px;
  font-weight: 500;
  font-style: italic;
}

.current-class-footer {
  padding: 80px 200px 30px;
  display: flex;
  justify-content: space-between;
}

.your-activities {
  line-height: 2em;
}

.button {
  border-radius: 40px;
  background-color: #AC00CB;
  padding: 10px 40px;
  color: white;
  margin-bottom: 20px;
  font-weight: 500;
}

.button:hover {
  cursor: pointer;
}

.App-link {
  color: #61dafb;
}

@keyframes App-logo-spin {
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
}


.Quiz-header {
  background-color: #222;
  padding: 10px;
  color: white;
  text-align: center;
}

.Quiz-intro {
  font-size: large;
}

@keyframes spin {
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
}