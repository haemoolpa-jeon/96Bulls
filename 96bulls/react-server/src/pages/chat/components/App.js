import React, { Component } from 'react';
import styles from '../style/Home.module.css';
import '../styles/main.scss';

import { connect } from 'react-redux';
import PropTypes from 'prop-types';

import { Link } from 'react-router-dom';
import userTypeStore from '../userTypeStore';

export class App extends Component {
  render() {
    return (
      <div className={styles.App}>
        <div className={styles.container}>
          <title>Altis Academia</title>
        <main className={styles.main}>
          <h1 className={styles.title}>
            Welcome to UQ remote
          </h1>
  
          <p className={styles.description}>
            Powered By Altis Academia
          </p>

          <div className={styles.chooseUserPage}>
            <div className={styles.chooseUserCard}>
              <div className='flex-center' style={{ backgroundColor: '#BF40FF', width: '100%', height: 60, color: 'white' }}>
                Choose user type
              </div>
              <div className="button" style={{ width: '200px', marginTop: 20 }} onClick={() => { userTypeStore.changeUserType('instructor');}}>
                <Link to = "/home">Instructor</Link>
              </div>
              <div className="button" style={{ width: '200px' }} onClick={() => { userTypeStore.changeUserType('student'); }}>
                <Link to = "/home">Student</Link>
              </div>
            </div>
          </div>          
          
        </main>
  
        <footer className={styles.footer}>
          <h2>Powered By Altis Academia</h2>
        </footer>
      </div>
      </div>
    );
  }
}

const mapStateToProps = state => ({
  currentUser: state.user.currentUser,
  currentRoom: state.room.currentRoom,
});

App.propTypes = {
  currentUser: PropTypes.object,
  currentRoom: PropTypes.object,
};

export default connect(mapStateToProps)(App);
