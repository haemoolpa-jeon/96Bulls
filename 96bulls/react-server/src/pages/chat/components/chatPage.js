import React from 'react';
import Chat from './chat'

class chatPage extends React.Component {

    render() {
        return (
            <div className="current-class-page">
                <div className='back-button'><a href='/'>‚üµ   Back</a></div>
                <Chat></Chat>
                </div>
        );
    }
}

export default chatPage;
