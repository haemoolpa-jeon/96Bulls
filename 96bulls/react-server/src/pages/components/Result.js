/* Results page of quiz when completed, show EXP and levels */

import React from 'react';
import PropTypes from 'prop-types';
import { CSSTransition } from 'react-transition-group';
import { useHistory } from 'react-router-dom'

function Result(props) {

  const history = useHistory();

  return (
    <CSSTransition
      className="container result"
      component="div"
      transitionName="fade"
      transitionEnterTimeout={800}
      transitionLeaveTimeout={500}
      transitionAppear
      transitionAppearTimeout={500}
    >
      <div>
        {
          props.quizResult === 1
            ? <div>
              You got <strong>{props.quizResult}</strong> question correct
            </div>
            : <div>
              You got <strong>{props.quizResult}</strong> questions correct
          </div>
        }
        <button className="button home-button" onClick={() => { history.push('/') }}>Home</button>
      </div>
    </CSSTransition>
  );
}

Result.propTypes = {
  quizResult: PropTypes.string.isRequired
};

export default Result;
