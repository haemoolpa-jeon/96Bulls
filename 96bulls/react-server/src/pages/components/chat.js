import React, { Component } from 'react';
import SidePanel from './SidePanel/SidePanel';
import RoomList from './RoomList/RoomList';
import Messages from './Messages/Messages';
import styles from './App.module.scss';

import { connect } from 'react-redux';
import PropTypes from 'prop-types';

export class Chat extends Component {
  render() {
    const { currentUser, currentRoom } = this.props;
    return (
      <div className={styles.App}>
        <SidePanel
          key={currentUser && currentUser.id}
          currentUser={currentUser}
        />
        <RoomList
          currentUser={currentUser}
        />
        <Messages
          key={currentRoom && currentRoom.id}
          currentUser={currentUser}
          currentRoom={currentRoom}
        />
      </div>
    );
  }
}

const mapStateToProps = state => ({
  currentUser: state.user.currentUser,
  currentRoom: state.room.currentRoom,
});

Chat.propTypes = {
  currentUser: PropTypes.object,
  currentRoom: PropTypes.object,
};

export default connect(mapStateToProps)(Chat);

