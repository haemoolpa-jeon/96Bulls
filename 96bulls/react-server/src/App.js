import React from 'react';
import './App.css';
import './style/page8.css'
import './style/leaderboard.css'
import './style/page5.css'
import './style/Home.module.css'
import 'bootstrap/dist/css/bootstrap.css'

import FrontPage from './pages/FrontPage.js';
import CurrentClass from './pages/CurrentClass.js';
import Profile from './pages/Profile.js';
import Leaderboard from './pages/Leaderboard.js';
import Page8 from './pages/Page8.js';
import QuizPage from './pages/QuizPageAlt.js';
import HomePage from './pages/HomePage.js';
import CreateQuiz from './pages/CreateQuizAlt.js';
import Quizzes from './pages/Quizzes.js';
import ChooseUserTypePage from './pages/ChooseUserTypePage.js';
import SignIn from './pages/signin.js';
import Register from './pages/register.js';

import { Route, BrowserRouter as Router, Switch } from 'react-router-dom'
import EditProfile from './pages/Canvas.js';


class App extends React.Component {

  render() {
    return (
      <Router>
        <Switch>
          <Route exact path="/" render={() => <ChooseUserTypePage />} />
          <Route exact path="/home" render={() => <HomePage />} />
          <Route exact path="/register" render={() => <Register />} />
          <Route exact path="/signin" render={() => <SignIn />} />
          <Route exact path="/class" render={() => <FrontPage />} />
          <Route exact path="/currentClass" render={() => <CurrentClass />} />
          <Route exact path="/profile" render={() => <Profile name="Christopher Walken" degree="Bachelor of Copy Paste" />} />
          <Route exact path="/editprofile" render={() => <EditProfile />} />
          <Route exact path="/leaderboard" render={() => <Leaderboard />} />
          <Route exact path="/page8" render={() => <Page8 />} />
          <Route exact path="/Quiz/:id" render={({ match }) => <QuizPage match={match} />} />
          <Route exact path="/CreateQuiz" render={() => <CreateQuiz />} />
          <Route exact path="/quizzes" render={() => <Quizzes />} />
        </Switch>
      </Router>
    );
  }
}

export default App;
