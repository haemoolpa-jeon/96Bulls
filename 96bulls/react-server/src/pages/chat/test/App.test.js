import React from 'react';
import { App } from '../components/App';
import {shallow} from "enzyme/build";

describe('Register', () => {
  it('renders correctly', () => {
    const component = shallow(<App/>);
    expect(component).toMatchSnapshot();
  });
})
