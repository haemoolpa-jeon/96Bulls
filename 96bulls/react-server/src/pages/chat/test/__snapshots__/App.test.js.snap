// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Register renders correctly 1`] = `
<div
  className="App"
>
  <Connect(SidePanel) />
  <Connect(RoomList) />
  <Messages />
</div>
`;
